src/api/MetadataQuery.js

```diff
@@ -147,54 +158,84 @@ class MetadataQuery extends Base {
      * @param {Object} - metadata query response object
      * @return {Object} - flattened metadata query response object
      */
-    flattenMetdataQueryResponse = ({ entries, next_marker }: MetadataQueryResponse): FlattenedMetadataQueryResponse => {
+    flattenMetdataQueryResponse = (): FlattenedMetadataQueryResponse => {
+        const { entries, next_marker }: MetadataQueryResponse = this.metadataQueryResponse;
         return {
             items: entries.map<FlattenedMetadataQueryResponseEntry>(this.flattenResponseEntry),
             nextMarker: next_marker,
         };
     };
 
     /**
-     * @param {Object} response
+     * Success handler function storing flattened response in Cache
+     * @param {Object} templateSchemaResponse - metadata template schema response
      */
-    queryMetadataSuccessHandler = ({ data }: { data: MetadataQueryResponse }): void => {
+    successHandler = (templateSchemaResponse: ?MetadataTemplateSchemaResponse): void => {
+        this.metadataTemplate = getProp(templateSchemaResponse, 'data');
```

Comment:
I think our eventual goal is for our API layer to be as stateless as possible. Can we avoid storing the response payload in the API class? Also, do we need `getProp` here or can we destructure this value in the function signature?

Reply:
We always store references to success and error callbacks, error codes at class level. This is to avoid passing/relaying arguments between methods. `getProp` is used to gracefully handle the absence of template (when `response.entries = [], we don't need to get template info`). Destructuring data in function signature won't work when templateSchemaResponse is `undefined`.

Reply:
We currently store callbacks and error codes because they tie into underlying functionality in the API base classes. Even in those cases we should aim to move away from it in the future. Adding _more_ state to the class in the form of API responses seems counterproductive.